opensrp-giz-malawi/src/main/java/org/smartregister/giz/dao/ReportDao.java

```diff
@@ -0,0 +1,421 @@
+        if (TextUtils.isEmpty(curVaccine.vaccine_separator)) {
+            String vaccineName = curVaccine.name;
+            VaccineSchedule vaccineSchedule;
+            if (curVaccine.schedule != null) {
+                vaccineSchedule = getVaccineSchedule(vaccineName, vaccineCategory, curVaccine.schedule);
+                vaccineSchedules.get(vaccineCategory).put(vaccineName.toUpperCase(), vaccineSchedule);
+            }
+        } else {
+            String[] splitNames = curVaccine.name
+                    .split(curVaccine.vaccine_separator);
+            for (int nameIndex = 0; nameIndex < splitNames.length; nameIndex++) {
+                String vaccineName = splitNames[nameIndex];
+                VaccineSchedule vaccineSchedule = getVaccineSchedule(vaccineName, vaccineCategory, curVaccine.schedules.get(vaccineName));
+                if (vaccineSchedule != null) {
+                    vaccineSchedules.get(vaccineCategory).put(vaccineName.toUpperCase(), vaccineSchedule);
+                }
+            }
+        }
+    }
+
+    private static HashMap<String, HashMap<String, VaccineSchedule>> getSchedule(
+            List<VaccineGroup> vaccines,
+            List<org.smartregister.immunization.domain.jsonmapping.Vaccine> specialVaccines,
+            String vaccineCategory
+    ) {
+        if (vaccineSchedules == null) {
+            vaccineSchedules = new HashMap<>();
+        }
+        vaccineSchedules.put(vaccineCategory, new HashMap<>());
+
+        for (VaccineGroup vaccineGroup : vaccines) {
+            for (org.smartregister.immunization.domain.jsonmapping.Vaccine vaccine : vaccineGroup.vaccines) {
+                initVaccine(vaccineCategory, vaccine);
+            }
+        }
+
+        if (specialVaccines != null) {
+            for (org.smartregister.immunization.domain.jsonmapping.Vaccine vaccine : specialVaccines) {
+                initVaccine(vaccineCategory, vaccine);
+            }
+        }
+
+        return vaccineSchedules;
+    }
+
+    private static List<Alert> getInMemoryAlerts(
+            HashMap<String, HashMap<String, VaccineSchedule>> vaccineSchedules,
+            String baseEntityId,
+            DateTime dob,
+            String vaccineCategory,
+            List<Vaccine> issuedVaccines
+    ) {
+        List<Alert> generatedAlerts = new ArrayList<>();
+        try {
+            if (vaccineSchedules != null && vaccineSchedules.containsKey(vaccineCategory)) {
+                for (VaccineSchedule curSchedule : vaccineSchedules.get(vaccineCategory).values()) {
+                    if (curSchedule == null) continue;
+                    Alert curAlert = curSchedule.getOfflineAlert(baseEntityId, dob.toDate(), issuedVaccines);
+                    if (curAlert != null && curAlert.startDate() != null) {
+                        generatedAlerts.add(curAlert);
+                    }
+                }
+            }
+            return generatedAlerts;
+        } catch (Exception e) {
```

Comment:
Could you catch a less generic exception?

Reply:
done